### ISSUE TITLE
Testathon: Submitter's name replaced with NA in workflow and elsewhere

### ISSUE THREAD

**Issue by librariankimberly** [DSpace/dspace-angular#2928]
The submitter's name is replaced with NA. This is probably a feature and not a bug, but how does this get toggled on and off? This question was brought forward by a member of the Brazil User Group and Pierre kindly took a screen print to save me time. (Thus the French). While I'm sure there are use cases to hide the submitter's name, many of us need to see the submitter's name when evaluating workflow requests in our live repositories.

![na in workflow](https://github.com/DSpace/dspace-angular/assets/42072233/41503cc8-24c3-4e4b-bfc8-0aeaa1761a29)

**Comment by alexandrevryghem** [comment]
@librariankimberly: This is on purpose it's was implemented during #2129 and is only displayed when the submitter's account has been removed

**Comment by IgorBaptist4** [comment]
Hello, this question was brought by me to Kimberly, so I will explain more about the behavior. I'm not able to reproduce it in the sandbox environment, but I'm able to reproduce it locally with a fresh install  (without change any configuration) (in versions 7.6, 7.6, and 8.0 from the dspace-8.0-testathon branch) which is something weird I only be able to reproduce it locally, looks like some configuration that was made in the sandbox change the behavior on this part of DSpace.

So, to do this test I created 3 accounts:

one account to submit the item, added only in the submitter group of the collection (submitter@neki-it.com.br) 

one account to do the test, checking the item submitter in MyDSpace, added only in the workflow editor group (workfloweditor@neki-it.com.br)

another account to do the test, checking the item submitter in MyDSpace, added in the workflow editor group and collection admin group (colladmin@neki-it.com.br)

Using the account that is only workflow editor, I'm not able to see the submitter of the item in MyDSpace:
![image](https://github.com/DSpace/dspace-angular/assets/68011073/62e2be13-fce3-4f66-89fa-979db5a11bf4)

I'm only able to see the submitter of the item, if I click on "View" and check the metadata dc.description.provenance of the item:
![image](https://github.com/DSpace/dspace-angular/assets/68011073/7f9ade47-7157-4fd8-9f24-5330dbda35e8)


When I'm using the account that is workflow editor + collection admin (or system admin), I'm able to see the submitter in MyDSpace:
![image](https://github.com/DSpace/dspace-angular/assets/68011073/40556509-8303-4666-8ee7-f8ba5bbb8ed9)

I tried to search for some configuration that add more permissions to workflow editor that can cause this different behavior between the sandbox environment and the fresh install locally, but I don't found nothing.

Thanks!
